moo.js

```diff
@@ -116,6 +123,14 @@
       // convert to RegExp
       var pat = pattern(obj.match)
 
+      if (pat === ERROR || (Array.isArray(obj) && obj.match.indexOf(ERROR) !== -1)) {
+        if (error) {
+          throw new Error("Cannot have multiple error rules")
```

Comment:
It's nice to provide context for the API user in the error message—the name of the token that generated this error, and perhaps the name of the token already marked as `error` as well. You should also add a compiler test case to ensure that this error is thrown when appropriate, and that the message contains the correct information.